cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1038,6 +1043,71 @@ public void setServiceInstances(HttpRequest request, HttpResponder responder,
     }
   }
 
+  /*************************** Queues APIs **************************************/
+  @DELETE
+  @Path("/queues")
+  public synchronized void clearQueues(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId) {
+    // synchronized to avoid a potential race condition here:
```

Comment:
the race still exists if we're running in HA mode.  We aren't introducing a new problem here, so I would say to open a jira to address the race and add a todo here with the jira.